workflow/metrics/server.go

```diff
@@ -1,33 +1,65 @@
 package metrics
 
 import (
-	"context"
 	"fmt"
 	"net/http"
-
-	"github.com/argoproj/argo/config"
+	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
+
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 	log "github.com/sirupsen/logrus"
 )
 
 // RunServer starts a metrics server
-func RunServer(ctx context.Context, config config.PrometheusConfig, registry *prometheus.Registry) {
+func (m Metrics) RunServer(stopCh <-chan struct{}) {
```

Comment:
The metrics object is now it's own server. This makes the metrics more of a "service" that should work "out-of-the box"